### PR TITLE
core/vm: optimisation on RETURN and updated tests

### PR DESCRIPTION
**Pull request by obscuren**
* Quadratic tests un-skipped
* MemoryStress tests un-skipped

Improved VM `RETURN` opcode to always return a pointer to memory rather than a copy of the memory. See `GetPtr(offset, size)` in `vm/memory.go`

### CODE REVIEW

File: tests/vm/gh_test.go [hunk 1]
```diff
@@ -128,9 +128,15 @@ func RunVmTest(p string, t *testing.T) {
 			ret, logs, gas, err = helper.RunState(statedb, env, test.Transaction)
 		}
 
-		rexp := helper.FromHex(test.Out)
-		if bytes.Compare(rexp, ret) != 0 {
-			t.Errorf("%s's return failed. Expected %x, got %x\n", name, rexp, ret)
+		switch name {
+		// the memory required for these tests (4294967297 bytes) would take too much time.
+		// on 19 May 2015 decided to skip these tests their output.
```
Comment (Gustav-Simonsson) [inline tests/vm/gh_test.go]: clarify that it's the validation of the returned value which would take too much time and memory

File: tests/vm/gh_test.go [hunk 2]
```diff
@@ -272,11 +277,13 @@ func TestVmLog(t *testing.T) {
 }
 
 func TestInputLimits(t *testing.T) {
+	t.Skip() // 1413631 lines of test code .. @jeff
```
Comment (Gustav-Simonsson) [inline tests/vm/gh_test.go]: These two run in a couple of seconds in develop, should be OK to leave unskipped?

Comment (obscuren) [inline tests/vm/gh_test.go]: This was due a left over of `runtime.GC` after each `vm.Run`

**Comment by obscuren** [comment]
This PR will fail on the build servers due to using too much memory in the state memory stress test and quadratic test. Is anyone aware of some conditional `t.Skip()`?

**Comment by fjl** [comment]
```go
if os.Getenv("SKIP_THE_TEST") != "" {
    t.Skip()
}
```

**Comment by fjl** [comment]
Or add [gosigar](https://github.com/cloudfoundry/gosigar) and do

```go
if sigar.Mem().Total < 2147483648 {
    t.Skipf("less than 2G of memory available")
}
```

**Comment by obscuren** [comment]
It's not travis i'm worried about.

**Comment by obscuren** [comment]
@fjl indeed. `TEST_VM_COMPLEX` will tests complex tests in the vm now.

**Pull request merged**